backend/src/main/java/com/github/NicolaiKopka/web_page_services/main_page/MainPageService.java

```diff
@@ -22,7 +24,49 @@ public List<Movie> getPopularMovies() {
         return movieDBApiConnect.getPopularMovies();
     }
 
-    public void getSoundtrackOnSpotify() throws OAuthProblemException, OAuthSystemException {
-        spotifyApiConnect.getSpotifyStatusForMovie();
+    public StreamingStatusDTO getSoundtrackOnSpotify(String movieName) throws OAuthProblemException, OAuthSystemException {
+        StreamingStatusDTO streamingStatusDTO = new StreamingStatusDTO();
+        streamingStatusDTO.setMovieName(movieName);
+
+        try {
+            List<SpotifyAlbum> albums = spotifyApiConnect.getSpotifyListOfMovieAlbums(movieName);
+
+            SpotifyAlbum album = checkForExactSoundtrack(albums, movieName);
+            String link = album.getExternalURLs().getAlbumUrl();
+            streamingStatusDTO.getStreamingServiceStatus().put("spotify", true);
+            streamingStatusDTO.getAlbumLinks().put("spotify", link);
+
+        } catch (Exception e) {
+            streamingStatusDTO.getStreamingServiceStatus().put("spotify", false);
+        }
+
+        return streamingStatusDTO;
+    }
+
+    private SpotifyAlbum checkForExactSoundtrack(List<SpotifyAlbum> albumList, String movieName) {
+        for(SpotifyAlbum album: albumList) {
+            if(checkForExactTitle(album.getName(), movieName) && checkForKeywords(album)) {
+                return album;
+            }
+        }
+        return null;
+
+
+    }
+    private boolean checkForKeywords(SpotifyAlbum album) {
+        return album.getName().toLowerCase().contains("official") || album.getName().toLowerCase().contains("motion picture")
+                || album.getName().toLowerCase().contains("soundtrack") || album.getName().toLowerCase().contains("netflix");
+    }
+    private boolean checkForExactTitle(String albumTitle, String movieName){
+        String albumTitleLower = albumTitle.split("\\(")[0].toLowerCase();
+        if (albumTitleLower.toCharArray()[albumTitleLower.length() - 1] == ' ') {
```

Comment:
This could be `if (albumTitleLower.charAt(albumTitleLower.length() - 1) == ' ') {`